### DISCUSSION TITLE
NextJS 13 App Directory

### DISCUSSION THREAD

**Discussion by laurent512** [htmlstreamofficial/preline#39] [General]
Following the guide to enable preline JS, we have to include  the following useEffect at the root : 
```
useEffect(() => {
        import("preline");
    }, []);
```

This is only feasible when "use client" is added, so root layout becomes client components which I think is not ideal. 

Any alternative solution ?

**Comment by evanrosa** [comment]
Any solution to this?

**Comment by arshtec** [comment]
Preline is a great library, after doing a bunch of research on this. I hope this silence with this NextJS 13+ integration won't turn people off. We just need a solution to this issue so we can start using this, otherwise, wonderful library.

**Reply by evanrosa** [reply]
I solved this by creating a separate provider which I then imported into the layout. Sorry for the formatting. For some reason the code formatter isn't working.

Provider:
`"use client";
import { SessionProvider } from 'next-auth/react';
import { useEffect } from 'react';

export const Providers = ({ children }: Props) => {
  useEffect(() => {
    import('preline');
  }, []);
  return <SessionProvider>{children}</SessionProvider>;
};`

Layout:
`export default function RootLayout({
  children
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <body>
        <Providers>
          <MyUserContextProvider>
            <Navbar />
            {children}
            <Footer />
          </MyUserContextProvider>
        </Providers>
      </body>
    </html>
  );
}`

**Reply by lemonjp** [reply]
I still got this error on nextjs 13
```
Warning: useEffect must not return anything besides a function
```
